src/lib/base-command.js

```diff
@@ -118,13 +117,13 @@ class BaseCommand extends Command {
   }
 
   doLog(message, always = false) {
-    if (always || !this.flags.quiet) {
-      CliUx.ux.log(message);
+    if (always || !this.flags?.quiet) {
+      this.log(message);
     }
   }
 
   spinnerStart(message) {
-    if (!this.flags.quiet) {
+    if (!this.flags.quiet && !this.flags.json) {
```

Comment:
```suggestion
    if (!(this.flags.quiet || this.flags.json)) {
```

To me this is more easily understandable. But that may be different for you 🙂